src/multi.c

```diff
@@ -367,7 +387,7 @@ void touchWatchedKey(redisDb *db, robj *key) {
     /* Check if we are already watching for this key */
     listRewind(clients,&li);
     while((ln = listNext(&li))) {
-        watchedKey *wk = listNodeValue(ln);
+        watchedKey *wk = member2struct(watchedKey, node, ln);
```

Comment:
https://github.com/redis/redis/actions/runs/3536583000/jobs/5935744690#step:6:549

Reply:
it would be nice to have a `member2struct` macro that we can rely on elsewhere, but if we can't find a good way, and we're afraid from bugs resulting in undefined-behavior optimizations, maybe we'll just go back to casting and add a comment that this member must be the first one in the struct.

let's wait for a bit to see if anyone has an idea.

Reply:
We can use `offsetof()` from `stddef.h`. This is part of the standard lib just to avoid this UB :) Sanitizer should not complain after we change this. 

I think we can write member2struct like this:

```c
#define member2struct(struct_name, member_name, member_addr) \
            ((struct_name *)((char*)member_addr - offsetof(struct_name, member_name)))
```

1- We use `offsetof()` instead of `member_offset()`( we can delete this now).
2- See I've changed `(uint8_t*)` cast to `(char*)`.
    This does not matter much but according to standard, we are only allowed to cast pointers to its own type, `char*` and `void*`. Let's try to follow the rules.

Reply:
great! I tested your changes, it work

Reply:
yes, it seems using offsetof() from the standard library is the right thing to do

Reply:
@tezc are you sure offsetof is part of C99?

Reply:
@oranagra It is even part of C89 and never seen a `stddef.h` implementation without it. So, I think it is safe to use it.

Reply:
i must be old then... i remember having to define it myself, and IIRC i started programming in 93

Reply:
I think we all do this, I defined it myself for a long time until I figured out it's been part of the standard forever :)

Reply:
I did not start programming until 2016, lack of experience i thought that we have to define it by ourselves... Thank you @tezc  I know this right way of coding from your comment :)